src/main/java/software/amazon/smithy/lsp/project/ProjectManager.java

```diff
@@ -57,12 +75,13 @@ public Project getProject(String uri) {
         String path = LspAdapter.toPath(uri);
         if (isDetached(uri)) {
             return detached.get(uri);
-        }  else if (mainProject.smithyFiles().containsKey(path)) {
-            return mainProject;
-        } else {
-            // Note: In practice, this shouldn't really happen because the server shouldn't
-            //  be tracking any files that aren't attached to a project. But for testing, this
-            //  is useful to ensure that fact.
+        }  else  {
+            for (Project project : projects.values()) {
+                if (project.smithyFiles().containsKey(path)) {
+                    return project;
+                }
+            }
+
             return null;
```

Comment:
Should this log an error or warning of some sort that the uri could not be found?